scripts/plot_instanovo_version_improvement.py

```
@@ -0,0 +1,106 @@
+    # --- Panel 1: Paired dot plot (peptide AUC) ---
+    datasets, v1, v2 = load_instanovo_paired("peptide_precision_plot_data.csv")
+    improvement = v2 - v1
+
+    ax = axes[0]
+    order = np.argsort(improvement)
+    colors = ["#d62728" if imp < 0 else "#2ca02c" for imp in improvement[order]]
+    ax.barh(range(len(datasets)), improvement[order], color=colors, height=0.7, alpha=0.8)
+    ax.set_yticks(range(len(datasets)))
+    ax.set_yticklabels([datasets[i] for i in order], fontsize=5)
+    ax.axvline(0, color="black", linewidth=0.5)
+    ax.set_xlabel("AUC improvement (v1.2.2 − v1.1.2)")
```

Ruff (0.15.14)

[warning] String contains ambiguous `−` (MINUS SIGN). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)